packages/ui-components/src/components/QrCode/QrCodeVisual.tsx

```diff
@@ -0,0 +1,203 @@
+    setIsDownloading(false);
+  };
+
+  if (!isEnlarged) {
+    if(options.length > 1) {
+      return (
+        <Container className={className} maxWidth="sm">
+          <p style={{color: 'white', textAlign: 'center'}}>QR Code</p>
+          {
+            options.map(({ name, value }, index) => {
+              if(index > 5) {
+                return
+              }
+              return (
+                <Grid 
+                  container 
+                  style={{paddingBottom: 5}}
+                  alignContent="center"
+                >
+                  <Grid item alignItems='flex-end' xs={6}>
+                  <QrCodeImage
+                      qrCodeContents={value}
+                      humanReadableId={name}
+                      width={50}
+                      margin="auto"
+                    />
+                  </Grid> 
+                  <Grid item alignItems='flex-start' xs={6} style={{ justifyItems: 'center'}}>
+                    <Typography variant="body2" style={{color: 'white'}}>
+                      {name}
+                    </Typography>
+                  </Grid> 
+                </Grid>
+              )
+            }
+          )
+        }
+        </Container>
+      )
+    }
+    const option = options[0]
+    return (
+      <Container>
+        <p style={{color: 'white', textAlign: 'center'}}>QR Code</p>
+        <Button style={{position: 'absolute', top: '5px', right: '5px'}} >
+              <DownloadIcon onClick={() => downloadSelectedQrCodes()}/>
+        </Button>
+        <QrCodeImage
+          qrCodeContents={option.value}
+          humanReadableId={option.name}
+        />
+      </Container>
+    );
+  }
+
+  if (!isLoading && options.length === 0) {
+    return (
+      <Container className={className}>
+        <NoData viewContent={config} />
+      </Container>
+    );
+  }
+
+  if (error) {
+    return (
+      <Container className={className}>
+        <Error>{error}</Error>
+        <DialogActions>
+          <Button onClick={onClose}>Cancel</Button>
+        </DialogActions>
+      </Container>
+    );
+  }
+
+  return (
+    <Container className={className}>
+      <FormContainer>
+        <FormControl>
+          <FormGroup>
+            {options.map(({ name, value }) => (
+              <FormControlLabel
+                key={value}
+                control={
+                  <Checkbox
+                    checked={selectedQrCodes[value]}
+                    checkedIcon={<CheckboxIcon className="checkbox-icon" />}
+                    onChange={() => toggleSelectFile(value)}
+                    value={value}
+                  />
+                }
+                label={name}
+              />
+            ))}
+          </FormGroup>
+        </FormControl>
+      </FormContainer>
+      <DialogActions>
+        <Button onClick={onClose}>Cancel</Button>
+        <Button
+          color="primary"
+          onClick={() => downloadSelectedQrCodes()}
```

Comment:
This doesn't need to be an inline function, and there are performance improvements when we don't have inline functions